enterprise_access/apps/provisioning/api.py

```diff
@@ -55,39 +60,58 @@ def get_or_create_enterprise_admin_users(enterprise_customer_uuid, user_emails):
         existing_admin_emails,
     )
 
-    existing_pending_admins = client.get_enterprise_pending_admin_users(enterprise_customer_uuid)
-    existing_pending_admin_emails = {record['user_email'] for record in existing_pending_admins}
-    logger.info(
-        'Provisioning: customer %s has existing pending admin emails %s',
-        enterprise_customer_uuid,
-        existing_pending_admin_emails,
-    )
-
     user_emails_to_create = list(
-        (set(user_emails) - existing_admin_emails) - existing_pending_admin_emails
+        set(user_emails) - existing_admin_emails
     )
 
     created_admins = []
     for user_email in user_emails_to_create:
-        result = client.create_enterprise_admin_user(enterprise_customer_uuid, user_email)
-        created_admins.append(result)
-        logger.info(
-            'Provisioning: created admin %s for customer %s',
-            user_email,
-            enterprise_customer_uuid,
-        )
+        try:
+            result = client.create_enterprise_admin_user(enterprise_customer_uuid, user_email)
+            created_admins.append(result)
+            logger.info(
+                'Provisioning: created admin %s for customer %s', user_email, enterprise_customer_uuid,
+            )
+        except APIClientException as exc:
+            if exc.__cause__.response.status_code == status.HTTP_404_NOT_FOUND:  # pylint: disable=no-member
+                result = _try_create_pending_admin(client, enterprise_customer_uuid, user_email)
+                if result:
+                    # AED 2025-10-14:
+                    # TODO: the endpoint to create real admins doesn't currently return the email address in the
```

Comment:
I think your comment meant to say "pending"
```suggestion
                    # TODO: the endpoint to create pending admins doesn't currently return the email address in the
```

Reply:
ahh, this comment and logic is in the wrong place - it really is the concrete admin creation response that doesn't contain in the email address in its payload.